client/asset/btc/btc.go

```diff
@@ -4355,6 +4363,47 @@ func (btc *baseWallet) createWitnessSig(tx *wire.MsgTx, idx int, pkScript []byte
 	return sig, privKey.PubKey().SerializeCompressed(), nil
 }
 
+// ValidateAddress checks that the provided address is valid.
+func (btc *baseWallet) ValidateAddress(address string) bool {
+	_, err := btcutil.DecodeAddress(address, btc.chainParams)
+	return err == nil
+}
+
+// EstimateSendTxFee returns a tx fee estimate for sending or withdrawing the
+// provided amount using the provided feeRate.
+func (btc *baseWallet) EstimateSendTxFee(address string, sendAmount, feeRate uint64, subtract bool) (fee uint64, isValidAddress bool, err error) {
+	if sendAmount == 0 {
+		return 0, false, fmt.Errorf("cannot check fee: send amount = 0")
+	}
+
+	isValidAddress = btc.ValidateAddress(address)
+
+	// Use dummy address for the estimation even if an address is provided. This
+	// ensures consistent fee estimate.
+	dummyP2WPKHAddr := "bc1qq49ypf420s0kh52l9pk7ha8n8nhsugdpculjas"
+
+	// This is just an estimate and it's safe to ignore err.
+	addr, _ := btcutil.DecodeAddress(dummyP2WPKHAddr, btc.chainParams)
+
+	pkScript, err := txscript.PayToAddrScript(addr)
+	if err != nil {
+		return 0, isValidAddress, fmt.Errorf("error generating pubkey script: %w", err)
+	}
```

Comment:
> If the provided address is valid, we should use it for the fee calc.

If the user decides to manually type the `Max Estimated Send` value into the amount field, there will be an error because the fee estimated with the dummyPkScript is will not be the same as the fee estimated with the provided address.

We can stick to the dummyPkscript and be consistent with the fee estimated.

Reply:
That makes it consistently *wrong*